devicetypes/smartthings/zigbee-motion-sensor-light.src/led-cpx-light.groovy

```diff
@@ -0,0 +1,166 @@
+	tiles(scale: 2) {
+		multiAttributeTile(name: "switch", type: "lighting", width: 6, height: 4, canChangeIcon: true) {
+			tileAttribute("device.switch", key: "PRIMARY_CONTROL") {
+				attributeState "on", label: '${name}', action: "switch.off", icon: "st.switches.light.on", backgroundColor: "#00A0DC", nextState: "turningOff"
+				attributeState "off", label: '${name}', action: "switch.on", icon: "st.switches.light.off", backgroundColor: "#ffffff", nextState: "turningOn"
+				attributeState "turningOn", label: '${name}', action: "switch.off", icon: "st.switches.light.on", backgroundColor: "#00A0DC", nextState: "turningOff"
+				attributeState "turningOff", label: '${name}', action: "switch.on", icon: "st.switches.light.off", backgroundColor: "#ffffff", nextState: "turningOn"
+			}
+			tileAttribute("device.level", key: "SLIDER_CONTROL") {
+				attributeState "level", action: "switch level.setLevel"
+			}
+		}
+		
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", label: "", action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+		
+		controlTile("colorTempSliderControl", "device.colorTemperature", "slider", width: 4, height: 2, inactiveLabel: false, range: "(2700..6500)") {
+			state "colorTemperature", action: "color temperature.setColorTemperature"
+		}
+		
+		main(["switch"])
+		details(["switch", "switchLevel", "colorTempSliderControl", "refresh"])
+	}
+}
+
+private getMOTION_CLUSTER_VALUE() { 0x0406 }
+private getMOTION_STATUS_VALUE() { 0x0000 }
+private getON_OFF_CLUSTER_VALUE() { 0x0006 }
+private getON_OFF_COMMAND_VALUE() { 0x07 }
+
+def parse(String description) {
+	def event = zigbee.getEvent(description)
+	
+	if (event) {
+		if (!(event.name == "level" && event.value == 0)) {
+			sendEvent(event)
+		}
+	} else {
+		def cluster = zigbee.parse(description)
+		
+		if (cluster && cluster.clusterId == ON_OFF_CLUSTER_VALUE && cluster.command == ON_OFF_COMMAND_VALUE) {
+			if (cluster.data[0] == 0x00) {
+				sendEvent(name: "checkInterval", value: 60 * 12, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+			}
+		} else {
+			def zigbeeMap = zigbee.parseDescriptionAsMap(description)
+			
+			if (zigbeeMap.clusterInt ==  MOTION_CLUSTER_VALUE && zigbeeMap.attrInt == MOTION_STATUS_VALUE) {
+				def childDevice = getChildDevices()?.find {
+					it.device.deviceNetworkId == "${device.deviceNetworkId}:1" 
+				}
+				zigbeeMap.value = zigbeeMap.value.endsWith("01") ? "active" : "inactive"			
+				zigbeeMap.name = "motion"
+				childDevice.sendEvent(zigbeeMap)
+			}
+		}
+	}
+}
+
+def off() {
+	zigbee.off()
+}
+
+def on() {
+	zigbee.on()
+}
+
+def setLevel(value, rate=null) {
+	zigbee.setLevel(value)
+}
+
+def configure() {
+	zigbee.configureReporting(MOTION_CLUSTER_VALUE, MOTION_STATUS_VALUE, 0x18, 30, 600, null) +
+		zigbee.onOffConfig() +
+		zigbee.levelConfig() +
+		refresh()
+}
+
+def updated() {
+	refresh()
```

Comment:
Perhaps an "emergency childDevice creation" would be a better use of `updated()` than calling `refresh()`.
```
if (!childDevices) {
	addChildSensor()
}
```

Reply:
OK, I changed that.